lib/screens/home_screen.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:first_project/extensions/context_extensions/colors.dart';
+import 'package:first_project/screens/login_screen.dart';
+import 'package:first_project/ui_components/shareable/add_button.dart';
+import 'package:first_project/ui_components/bars/chip_bar.dart';
+import 'package:first_project/ui_components/bars/my_app_bar.dart';
+import 'package:first_project/ui_components/bars/my_search_bar.dart';
+import 'package:first_project/ui_components/user_tiles.dart';
+import 'package:flutter/material.dart';
+
+class HomeScreen extends StatefulWidget {
+  static const routeName = '/home';
+
+  const HomeScreen({super.key});
+
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> with WidgetsBindingObserver {
+  late DateTime markTime;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    super.didChangeAppLifecycleState(state);
+    if (state == AppLifecycleState.paused) markTime = DateTime.now();
+    if (state == AppLifecycleState.resumed && (DateTime.now().difference(markTime) >= const Duration(minutes: 1))) {
+      Navigator.pushReplacementNamed(context, LoginScreen.routeName);
+    }
```

Comment:
This works - nice job

It is better tho to separate this out into a notifier. One flaw of current solution is that home screen state could be disposed for some reason.